### DISCUSSION TITLE
When to use Magick.NET & usage question

### DISCUSSION THREAD

**Discussion by xMaxximum** [dlemstra/Magick.NET#1485] [Help] [answered]
Hello,
I'm having trouble understanding when one should use Magick.NET and when one should just use the Graphics builtin library.

Additionally, I would like to ask how I could programmatically, given an input of a background color and a list of strings, be able to make it automatically calculate the needed font size as well as keep a margin so that it still fits and looks nicely. 
Basically, get from base to goal. Font is "IFC RAILROAD".

base:
![base](https://github.com/dlemstra/Magick.NET/assets/64035335/27540e22-9b36-46b5-8146-c0fd8b882707)
goal:
![goal](https://github.com/dlemstra/Magick.NET/assets/64035335/057bffe6-bd1d-4907-8d5a-b4dfaa31bf5f)

Any help and pointing to resources are appreciated.

Thanks

**Comment by dlemstra** [comment, marked as answer]
Your first question is not easy to answer. It all depends on what else you want to do or which kinds of formats you want to support so I don't have an answer to that question. But using my library could be one of your options.

And if you use this library you can read an image with a label that fits in the specified dimensions like this:

```C#
var readSettings = new MagickReadSettings
{
    BackgroundColor = MagickColors.Transparent,
    FillColor = new MagickColor("#bebcbe"),
    TextGravity = Gravity.Center,
    FontFamily = "IFC RAILROAD", // Not sure if this is the correct font name
    // Font = @"c:\path\font.ttf", // This might be an other option to specify the font.
    Width = 200, // This should also be changed to the box that you now have.
    Height = 100,
};

using var image = new MagickImage("label:MEMBER", readSettings);
```

And after you have read the image you can use the `Compose` method of `image` to draw this image but I will leave that as an exercise for the reader. But feel free to ask more questions if you need some more information.

**Comment by xMaxximum** [comment]
Hello,
thanks for your quick response! I apologize because I have been busy.

```cs
            var readSettings = new MagickReadSettings()
            {
                BackgroundColor = MagickColors.Transparent,
                FillColor = new MagickColor("#bebcbe"),
                TextGravity = Gravity.Center,
                FontPointsize = 200,
                Font = @"IFC RAILROAD.ttf",
                Width = 2000,
                Height = 900
            };

            using var img = new MagickImage("base.png");

            using var caption = new MagickImage("label:MEMBER", readSettings);

            img.Composite(caption, 500, 400, CompositeOperator.Over);

            img.Write("output.png");
```

but I still don't understand how I could automatically read every box inside the base image that consists of the light gray background in which the text should be written to. Could you tell me how to realize that?

**Reply by dlemstra** [reply]
You cannot read the box you will need to supply that information yourself? You will need to specify the location of that box and the size of that box.

**Reply by xMaxximum** [reply]
Okay I see, thanks!
